app/services/the_ingestor.rb

```diff
@@ -8,22 +8,26 @@ def ingest
     ensure_collection_exists
 
     @document.chunking!
-    @chunks = chonker.chunks.map { |chunk| Chunk.new(document: @document, content: chunk) }
-    Rails.logger.info("Got #{@chunks.count} chunks from #{@document.filename}.")
-
-    ## save all chunks to the db
-    @document.transaction do
-      @chunks.each(&:save!)
-    end
-    @document.chunked!
-
-    @document.embedding!
-    @chunks.each_with_index do |chunk, idx|
-      Rails.logger.info("Embedding chunk #{chunk.id} (#{idx}/#{@chunks.count})...")
+    index = 0
+    chonker.each do |c|
```

Comment:
```suggestion
    chonker.each_with_index do |c, index|
```

Reply:
Since `Chonker` is not implementing `Enumerator`, I thought this wouldn't work. I.e. Only `#each` is available from `Chonker`.

Reply:
True. You could just implement `each_with_index` to avoid this C code. ;)

Reply:
Done ... for now. :-)